aggregator_core/src/datastore.rs

```diff
@@ -2036,59 +1967,52 @@ impl<C: Clock> Transaction<'_, C> {
         role: &Role,
         task_id: &TaskId,
         aggregation_job_id: &AggregationJobId,
-        aggregation_param: &A::AggregationParam,
-        report_id: &ReportId,
-    ) -> Result<Option<ReportAggregation<SEED_SIZE, A>>, Error>
+    ) -> Result<Vec<ReportAggregation<SEED_SIZE, A>>, Error>
     where
         for<'a> A::PrepareState: ParameterizedDecode<(&'a A, usize)>,
     {
         let stmt = self
             .prepare_cached(
                 "SELECT
-                    report_aggregations.client_timestamp, report_aggregations.ord,
-                    report_aggregations.last_prep_resp,
-                    report_aggregations.state,
-                    report_aggregations.public_share, report_aggregations.leader_extensions,
-                    report_aggregations.leader_input_share,
-                    report_aggregations.helper_encrypted_input_share,
-                    report_aggregations.leader_prep_transition,
-                    report_aggregations.helper_prep_state, report_aggregations.error_code
+                    ord, client_report_id, client_timestamp, last_prep_resp,
+                    report_aggregations.state, public_share, leader_extensions, leader_input_share,
+                    helper_encrypted_input_share, leader_prep_transition, helper_prep_state,
+                    error_code
                 FROM report_aggregations
                 JOIN aggregation_jobs ON aggregation_jobs.id = report_aggregations.aggregation_job_id
                 JOIN tasks ON tasks.id = aggregation_jobs.task_id AND tasks.id = report_aggregations.task_id
                 WHERE tasks.task_id = $1
                   AND aggregation_jobs.aggregation_job_id = $2
-                  AND report_aggregations.client_report_id = $3
-                  AND UPPER(aggregation_jobs.client_timestamp_interval) >= COALESCE($4::TIMESTAMP - tasks.report_expiry_age * '1 second'::INTERVAL, '-infinity'::TIMESTAMP)",
+                  AND UPPER(client_timestamp_interval) >= COALESCE($3::TIMESTAMP - tasks.report_expiry_age * '1 second'::INTERVAL, '-infinity'::TIMESTAMP)
+                ORDER BY ord ASC",
```

Comment:
I wonder if for `get_report_aggregations_for_aggregation_job()` we should take the similar approach to splitting the query as https://github.com/divviup/janus/pull/2807.

On one hand, without a query plan against a large enough table this could be premature optimization, OTOH we know postgres has a hard time planning this COALESCE statement.

Reply:
I agree. I think I'm going to extend the approach in #2807 to every relevant query (and maybe add caching), but in a separate PR.